### ISSUE TITLE
Mysensors Wifi gateway reset on gw version request

### ISSUE THREAD

**Issue by ciprianpop3scu** [domoticz/domoticz#231]
I use Domoticz beta ver. 2.3708 with Ethernet MySensors Hardware. Before updating to latest version all was ok. Now my gateway is reseting when Domoticz is requesting version:
Gateway startup complete.
Client 0: 0;0;3;0;2;

Exception (28):
epc1=0x4000bf80 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

ctx: cont 
sp: 3ffebf60 end: 3ffec1f0 offset: 01a0

> > > stack>>>
> > > 3ffec100:  00000000 00000000 3ffec12e 4020102c  

  I have tried same request from telnet client:
Client 0: 0;0;3;0;2;
0;0;3;0;2;1.5
Client 0: 0;0;3;0;2;
0;0;3;0;2;1.5
  and it responded without crashing.

  I use ESP8266+NRF24 as a gateway (http://www.mysensors.org/build/esp8266_gateway).

  As a temporary solution I have drop from iptables sending packet with string '0;0;3;0;2;' to my gw and it is working for now:
Domoticz log
2015-11-28 12:58:05.489 (Ethernet MySensors) Temp (B)
2015-11-28 12:58:05.512 (Ethernet MySensors) General/Voltage (B)
2015-11-28 12:58:37.539 (Ethernet MySensors) Temp (B)
2015-11-28 12:58:37.563 (Ethernet MySensors) General/Voltage (B) 

Thanks

**Comment by gizmocuz** [comment]
The crash report is from your gateway ?
If it the default mysensors 1.6 ethernet gateway sketch?

**Comment by ciprianpop3scu** [comment]
yes, the crash is from my gateway and I use the default ESP8266 WiFi Gateway from Mysensors version 1.5, I know this is the last stable version.

**Comment by ciprianpop3scu** [comment]
if it helps, from a tcpdump i see this:
Domoticz -> gateway
data hex "303b303b333b303b323b0a" , asci "0;0;3;0;2;"
this chrashes my gateway

telnet -> gateway
data hex "303b303b333b303b323b0d0a" , asci "0;0;3;0;2;"
this is ok

the difference is the ending:
0a (hex for LF) vs 0d0a (hex for CR LF)

**Comment by gizmocuz** [comment]
I think you better report this also on the mysensors forum

**Comment by gizmocuz** [comment]
could be it is fixed in the latest 1.6 development branch

https://github.com/mysensors/Arduino/commit/f3b291f2e9085b8639a34ed2ac9d880d74f83316

**Comment by ciprianpop3scu** [comment]
I have added in the sketch example:
        if (inChar == '\n')
        {  
-          if (inputString[i].string[inputString[i].idx-1] != '\r') { // if the last char is not CR then add it
-          inputString[i].string[inputString[i].idx++]='\r';
- ```
       }
    // a command was issued by the client
    // we will now try to send it to the actuator
    inputString[i].string[inputString[i].idx] = 0;
  
    // echo the string to the serial port
    Serial.print("Client "); Serial.print(i); Serial.print(": "); Serial.println(inputString[i].string);
  
    parseAndSend(gw, inputString[i].string);
  ```
  
  and now it is working.
  I think it is fixed in the development branch.
  Thanks for your support, I will close the issue.
